Services/AspireService.cs

```diff
@@ -1,93 +1,119 @@
-                }
-                await responseStream.WriteAsync(logsUpdate);
-                // Introduce delay for testing
-                await Task.Delay(1000);
-            }
-        }
-
-        public override Task<ResourceCommandResponse> ExecuteResourceCommand(ResourceCommandRequest request, ServerCallContext context)
-        {
-            return base.ExecuteResourceCommand(request, context);
-        }
-
-    }
-}
+            Console.WriteLine($"WatchResources connection canceled");
+        }
+
+        private async Task<WatchResourcesUpdate> GetInitialData()
+        {
+            // Gets the initial data and return it
+            var podsList = await _kubernetesClient.CoreV1.ListNamespacedPodAsync(_kubernetesConfig.Namespace);
+            var pod = podsList.Items[0];
+            var initialData = new WatchResourcesUpdate()
+            {
+                InitialData = new InitialResourceData()
+                {
+                    ResourceTypes = { new ResourceType() { UniqueName = "Pod" } },
+                    Resources = { new Resource() { Name = "Pod", Endpoints = { new Aspire.V1.Endpoint() { EndpointUrl = pod.Status.PodIP } }, Commands = { new ResourceCommandRequest() { CommandType = "Restart" } } }
+
+                    }
+                }
+            };
+            return initialData;           
+        }
+
+        private async Task WatchAndWriteChangesToStream(IServerStreamWriter<WatchResourcesUpdate> responseStream)
+        {
+            // Creates the watcher
+            var podsWatchResponse = await _kubernetesClient.CoreV1.ListNamespacedPodWithHttpMessagesAsync(_kubernetesConfig.Namespace, watch: true);
+            var podWatcher = podsWatchResponse.Watch<V1Pod, V1PodList>(async (type, item) =>
+            {
+                
+                Console.WriteLine($"Pod event of type {type} detected for {item.Metadata.Name}");
+                var reply = new WatchResourcesUpdate()
+                {
+                    // TODO : Figure out deletion or upsert and add here.
+                    Changes = new WatchResourcesChanges()
+                    {
+                        Value = { new WatchResourcesChange() { Delete = new ResourceDeletion() { }, Upsert = new Resource { Name = item.Metadata.Name} } }
+                    }
+                    
+                };
+                await responseStream.WriteAsync(reply);
+            });
+        }
+
+        public override async Task WatchResourceConsoleLogs(WatchResourceConsoleLogsRequest request, IServerStreamWriter<WatchResourceConsoleLogsUpdate> responseStream, ServerCallContext context)
+        {
+           var label = $"app={request.ResourceName}";
+            _logger.LogInformation($"Label : {label}");
+           var pods = await _kubernetesClient.CoreV1.ListNamespacedPodAsync(_kubernetesConfig.Namespace, labelSelector: label);
+           var podName = pods.Items[0].Metadata.Name;

```

Comment:
Should we at least log if there is more than one pod? If we expect one pod to be present, then we should have an easy way to detect when this assumption is not true

Reply:
Added log for this in my latest commit.